doc/nrf/libraries/zigbee/osif.rst

```diff
@@ -40,6 +41,55 @@ Additionally, the following Kconfig option is available when setting :ref:`zigbe
 
 * :kconfig:`CONFIG_ZBOSS_OSIF_LOG_LEVEL` - Configures the custom logger options for the ZBOSS OSIF layer.
 
+.. _zigbee_osif_zboss_osif_serial:
+
+ZBOSS OSIF Serial
+*****************
+
+Setting the :kconfig:`CONFIG_ZIGBEE_HAVE_SERIAL` enables serial abstract for the ZBOSS OSIF.
+
+ZBOSS OSIF serial implements a set of functions used by ZBOSS stack for serial communication, from which two sets of functions with common purpose can be distinguish:
+
+* Handling NCP communication with HOST device - used only for :ref:`NCP architecture <ug_zigbee_platform_design_ncp>`
+* Logging ZBOSS Traces - stack logs useful for debugging, received in binary format.
+
+.. _zigbee_osif_zigbee_async_serial:
+
+ZBOSS OSIF serial supports single backend for handling NCP communication - Zigbee Async Serial.
+It uses :ref:`Zephyr UART API<uart_api>` and can be configured to use UART peripheral or USB CDC ACM device.
+Received data is internally buffered and if Logging ZBOSS Traces is supported, data to transmit is also buffered.
+To configure this, use the following options:
+
+* :kconfig:`CONFIG_ZIGBEE_HAVE_ASYNC_SERIAL` - to enable Zigbee Async Serial.
+* :kconfig:`CONFIG_ZIGBEE_UART_DEVICE_NAME` - to specify serial device to use.
+* :kconfig:`CONFIG_ZIGBEE_UART_SUPPORTS_FLOW_CONTROL` - set this option for serial devices that support flow control.
+* :kconfig:`CONFIG_ZIGBEE_UART_RX_BUF_LEN` - to configure size of internal RX and TX buffer.
+* :kconfig:`ZBOSS_TRACE_BINARY_NCP_TRANSPORT_LOGGING` - to enable Logging ZBOSS Traces in binary format with Zigbee Async Serial.
+
+Backend options for Logging ZBOSS Traces:
+
+* Zigbee Async Serial - allows to use the same backend for NCP communication and Logging ZBOSS Traces. Logs ZBOSS Traces in binary format.
+  For how to configure, see :ref:`Zigbee Async Serial<zigbee_osif_zigbee_async_serial>`.
+
+* Zigbee Serial Logger - uses :ref:`Zephyr UART API<uart_api>` and can be configured to use UART peripheral or USB CDC ACM device.
+  Data is buffered internally in ring buffer and printed in binary format. Use the following options to configure it:
+
+  * :kconfig:`CONFIG_ZBOSS_TRACE_BINARY_LOGGING` - to enable Logging ZBOSS Traces with Zigbee Serial Logger.
+  * :kconfig:`CONFIG_ZBOSS_TRACE_UART_LOGGING` - UART serial backend.
+  * :kconfig:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` - USB CDC ACM serial backend.
+  * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME` - to specify the serial device to use.
+  * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_BUFFER_SIZE` - to specify size of internal ring buffer.
+
+* Zigbee logger - uses Zephyr's :ref:`zephyr:logging_api` API for Logging hexdumps of received binary data.
+  Data is buffered internally in ring buffer. Use the following options to configure it:
```

Comment:
Zigbee logger configuration options
-------------------------------------

Use the following options to configure the Zigbee logger:

Reply:
will do